tarpc/src/rpc/client/channel.rs

```diff
@@ -24,11 +24,20 @@ use std::{
     io,
     pin::Pin,
     sync::{
-        atomic::{AtomicU64, Ordering},
+        atomic::{AtomicUsize, Ordering},
         Arc,
     },
 };
 
+#[allow(dead_code)]
+#[allow(clippy::no_effect)]
+const CHECK_USIZE: bool = {
+    if std::mem::size_of::<usize>() > std::mem::size_of::<u64>() {
+        ["usize is big small to fit in u64"][42];
```

Comment:
Nit: consider using the [compile_error](https://doc.rust-lang.org/std/macro.compile_error.html) macro.

Reply:
ah, I see. Ok, this works then! Perhaps add a TODO?